massa-versioning-worker/src/versioning.rs

```diff
@@ -0,0 +1,1057 @@
+use std::cmp::Ordering;
+use std::collections::{BTreeMap, BTreeSet};
+use std::hash::{Hash, Hasher};
+use std::ops::Deref;
+use std::sync::Arc;
+
+use machine::{machine, transitions};
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use parking_lot::RwLock;
+use thiserror::Error;
+
+use massa_models::{amount::Amount, config::VERSIONING_THRESHOLD_TRANSITION_ACCEPTED};
+use massa_time::MassaTime;
+
+// TODO: add more items here
+/// Versioning component enum
+#[allow(missing_docs)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash, TryFromPrimitive, IntoPrimitive)]
+#[repr(u32)]
+pub enum MipComponent {
+    Address,
+    Block,
+    VM,
+}
+
+/// MIP info (name,  version,
+#[derive(Clone, Debug)]
+pub struct MipInfo {
+    /// MIP name or descriptive name
+    pub name: String,
+    /// Network (or global) version
```

Comment:
Add some context (for other devs, to be able to differentiate with the `VERSION` const that is e.g. TEST.20), for example:

```suggestion
    /// Network (or global) version, to be included in block headers
```

Reply:
done